addons/website_sale/models/product_template.py

```diff
@@ -93,6 +93,8 @@ def _prepare_variant_values(self, combination):
 
     def _get_website_accessory_product(self):
         domain = self.env['website'].sale_product_domain()
+        if not self.env.user._is_internal():
+            domain = expression.AND([domain, [('is_published', '=', True)]])
         return self.accessory_product_ids.filtered_domain(domain)
 
     def _get_website_alternative_product(self):
```

Comment:
The same problem doesn't happen with alternative products ?

Reply:
I checked alternative and optional products and it was not happening, my guess is that it's solved in a different way.